addons/mrp/models/stock_move.py

```diff
@@ -135,15 +137,26 @@ def _compute_is_done(self):
         for move in self:
             move.is_done = (move.state in ('done', 'cancel'))
 
+    @api.depends('product_uom_qty')
+    def _compute_unit_factor(self):
+        for move in self:
+            mo = move.raw_material_production_id or move.production_id
+            if mo:
+                move.unit_factor = (move.product_uom_qty - move.quantity_done) / ((mo.product_qty - mo.qty_produced) or 1)
+            else:
+                move.unit_factor = 1.0
+
     @api.model
     def default_get(self, fields_list):
         defaults = super(StockMove, self).default_get(fields_list)
         if self.env.context.get('default_raw_material_production_id'):
             production_id = self.env['mrp.production'].browse(self.env.context['default_raw_material_production_id'])
-            if production_id.state == 'done':
-                defaults['state'] = 'done'
+            if production_id.state in ('confimed', 'done'):
+                defaults['state'] = production_id.state
                 defaults['product_uom_qty'] = 0.0
                 defaults['additional'] = True
+                defaults['group_id'] = production_id.procurement_group_id.id,
```

Comment:
if you're writing the state in this default_get, action_confirm won't apply pull and push won't it?